### PR TITLE
Helm chart with minikube tests (#432)

### PR DESCRIPTION
**Pull request by algattik**
A [Helm Chart](https://helm.sh) is provided to be used to deploy the EDC runtime in Kubernetes environments.

### Chart design

The chart is highly configurable:

- Configurable runtime based on Docker image of base EDC, Data Plane Framework Server, or custom runtime.
- Configurable environment variables (as [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/)) for runtime settings, following [12-factor principles](https://12factor.net/config).
- Configurable secrets (injected as environment variables into pods).
- Configurable labels and annotations.
- Configurable ingress.

The chart was generated using `helm create` in order to follow best practice for components, configurability and ingress definitions, and adapted to fit the EDC runtime.

A generic Dockerfile was created that builds a Docker image using any runtime JAR. It is designed to be used as-is and take configuration from environment variables. Alternatively, implementors can extend the Dockerfile to package a configuration file.

### Testing

The chart must be tested in CI, like any other code, to ensure it remains valid as either the chart or EDC code evolves.

A GitHub worklow installs minikube (a local lightweight kubernetes installation) and deploys two releases of the chart, for the Consumer and Provider EDC connector of sample 04.0. The workflow then triggers file copy and verifies that the destination file was created on the Provider pod.

### CODE REVIEW

**Review by jimmarino — APPROVED** [review]

File: launchers/generic/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+RUN apt-get update && apt-get install -y \
+  bash \
+  && rm -rf /var/lib/apt/lists/*
+
+COPY $JAR app.jar
+
+# Use "bash" (instead of default "sh") to support environment variables containing a dot. 
+# Use "exec" for Kubernetes graceful termination (SIGINT) to reach JVM.
+ENTRYPOINT [ "bash", "-c", \
+    "exec java $JVM_ARGS -jar app.jar"]
```
Comment (florianrusch-zf) [inline launchers/generic/Dockerfile:27]: Do we really have environment variables with a dot? Thought we have some kind mapping mechanism which transforms between screaming snake case and the dot case. E.g `STUFF_TEST` to `stuff.test`

Comment (algattik) [reply launchers/generic/Dockerfile:27]: This mapping exists only for settings fetched with `propOrEnv` e.g. the location of the config file `edc.fs.config`, not for other settings:

```
$ env web_http_port=9192 WEB_HTTP_PORT=9192 java -jar samples/04.0-file-transfer/consumer/build/libs/consumer.jar
INFO 2022-02-17T08:53:52.549321 HTTP context 'default' listening on port 8181


$ env web.http.port=9192 java -jar samples/04.0-file-transfer/consumer/build/libs/consumer.jar
INFO 2022-02-17T08:54:00.39245 HTTP context 'default' listening on port 9192
```

Comment (florianrusch-zf) [reply launchers/generic/Dockerfile:27]: Interesting. Do you or anyone else know if there is already a plan to align this to have some consistent ways to set the variables?

Comment (DominikPinsel) [reply launchers/generic/Dockerfile:27]: In issue #608 there is a discussion to make environment variables always snake case.

Comment (paullatzelsperger) [reply launchers/generic/Dockerfile:27]: I think the following should happen:
- internally in the runtime we only use the dot notation
- the `Config` mechanism checks for properties in the dot notation and env-vars in the uppercase-underscore notation
- `propOrEnv` should not be used anywhere but tests

Comment (ndr-brt) [reply launchers/generic/Dockerfile:27]: to be fair, that conversion was not in place before, it has only been implemented in the `propOrEnv`. However an issue about that was already opened: #608

Comment (florianrusch-zf) [reply launchers/generic/Dockerfile:27]: Thank you guys! Didn't want to start a discussion here, so lets move over with this topic to the other issue 😉

Comment (paullatzelsperger) [reply launchers/generic/Dockerfile:27]: doesn't matter now, i didn't want to assign blame :) Your comment on #608 is correct and should be done so.

Comment (jimmarino) [reply launchers/generic/Dockerfile:27]: How about fixing the setting properties by normalizing them as suggested by @ndr-brt and @paullatzelsperger and then realigning this PR with that?

Comment (jimmarino) [reply launchers/generic/Dockerfile:27]: I'm OK to merge this as-is and fix the other issue ASAP. We can then convert shells in a subsequent PR if wanted.

**Review by ndr-brt — COMMENTED** [review]
I would avoid the `charts` folder in an already crowded root folder, I would put it into `resources`

**Comment by paullatzelsperger** [comment]
Moving the `charts` folder to `resources` as @ndr-brt suggested seems reasonable and will probably have little impact on the rest of the code. Other than that I'm OK with this.

**Comment by algattik** [comment]
> I would avoid the `charts` folder in an already crowded root folder, I would put it into `resources`

Done. I would still recommend to put it in the root though, so that “casual” users find it

**Comment by ndr-brt** [comment]
> > I would avoid the `charts` folder in an already crowded root folder, I would put it into `resources`
> 
> Done. I would still recommend to put it in the root though, so that “casual” users find it

We can document that in the README, probably is the first place where a "casual" user take a look at.

**Comment by algattik** [comment]
> > > I would avoid the `charts` folder in an already crowded root folder, I would put it into `resources`
> > 
> > 
> > Done. I would still recommend to put it in the root though, so that “casual” users find it
> 
> We can document that in the README, probably is the first place where a "casual" user take a look at.

I have already :) thanks

**Review by paullatzelsperger — APPROVED** [review]

**Review by ndr-brt — APPROVED** [review]

**Pull request merged**